### PR TITLE
feat : Back End Notification Lab Account Verification

### PR DESCRIPTION
**Pull request by rubenkristian**
### JIRA LINK
- https://blocksphere2020.atlassian.net/browse/DBIO-2512
- https://blocksphere2020.atlassian.net/browse/DBIO-2513
- https://blocksphere2020.atlassian.net/browse/DBIO-2514
- https://blocksphere2020.atlassian.net/browse/DBIO-2515

### Changelog
- notification for verified, rejected, revoked
- notification verification reward after reward success
- update unit test

### CODE REVIEW

**Review by hilyds — COMMENTED** [review]
beside the 1 minor comment, it looks good to me @rubenkristian

**Review by hilyds — APPROVED** [review]

**Review by agustinustheo — APPROVED** [review]

File: src/endpoints/verification/verification.service.ts [hunk 1]
```diff
@@ -9,45 +9,89 @@ import {
...
+        const notificationInput: NotificationDto = {
+          role: 'Lab',
+          entity_type: 'Lab',
+          entity: '',
+          description: '',
+          read: false,
+          created_at: this.dateTimeProxy.new(),
+          updated_at: this.dateTimeProxy.new(),
+          deleted_at: null,
+          from: null,
```
Comment (hilyds) [inline src/endpoints/verification/verification.service.ts:42]: from: 'Debio Network',

File: src/endpoints/verification/verification.service.ts [hunk 2]
```diff
@@ -9,45 +9,89 @@ import {
...
+            role: 'Lab',
+            entity_type: 'Reward',
+            entity: 'Lab verified',
+            description:
+              'Congrats! You’ve got 2 DBIO from account verification.',
+            read: false,
+            created_at: this.dateTimeProxy.new(),
+            updated_at: this.dateTimeProxy.new(),
+            deleted_at: null,
+            from: null,
```
Comment (hilyds) [inline src/endpoints/verification/verification.service.ts:88]: from: 'Debio Network',

File: src/endpoints/verification/verification.service.ts [hunk 3]
```diff
@@ -9,45 +9,89 @@ import {
...
+    await updateLabVerificationStatus(
+      this.subtrateService.api as any,
+      this.subtrateService.pair,
+      substrateAddress,
+      <VerificationStatus>verificationStatus,
+      async () => {
+        // insert notification
+        const notificationInput: NotificationDto = {
+          role: 'Lab',
+          entity_type: 'Lab',
```
Comment (hilyds) [inline src/endpoints/verification/verification.service.ts:35]: entity_type: 'Verification',

File: src/endpoints/verification/verification.service.ts [hunk 4]
```diff
@@ -9,45 +9,89 @@ import {
...
+          created_at: this.dateTimeProxy.new(),
+          updated_at: this.dateTimeProxy.new(),
+          deleted_at: null,
+          from: null,
+          to: substrateAddress,
+        };
+
+        switch (verificationStatus) {
+          case VerificationStatus.Verified:
+            notificationInput.entity = 'Lab verified';
```
Comment (hilyds) [inline src/endpoints/verification/verification.service.ts:48]: notificationInput.entity = 'Account verified';

File: src/endpoints/verification/verification.service.ts [hunk 5]
```diff
@@ -9,45 +9,89 @@ import {
...
+            notificationInput.description =
+              'Congrats! Your account has been verified.';
+            break;
+          case VerificationStatus.Revoked:
+            notificationInput.entity = 'Lab revoked';
```
Comment (hilyds) [inline src/endpoints/verification/verification.service.ts:53]: notificationInput.entity = 'Account revoked';

File: src/endpoints/verification/verification.service.ts [hunk 6]
```diff
@@ -9,45 +9,89 @@ import {
...
+            notificationInput.description = 'Your account has been revoked.';
+            break;
+          case VerificationStatus.Rejected:
+            notificationInput.entity = 'Lab rejected';
```
Comment (hilyds) [inline src/endpoints/verification/verification.service.ts:57]: notificationInput.entity = 'Account rejected';

**Review by hilyds — CHANGES_REQUESTED** [review]

**Comment by rubenkristian** [comment]
image

**Pull request merged**